students/HarveyHerela/lesson07/html_render.py

```diff
@@ -0,0 +1,128 @@
+class Element:
+    """Base class for all HTML tags."""
+    def __init__(self, tag_name='element', content=None, **kwargs):
+        if content is None:
+            self.content = list()
+        else:
+            self.content = [content]
+        tag_style = ''
+        if 'style' in kwargs:
+            tag_style = ' style="{style}"'.format(style=kwargs['style'])
+        self.tag_open = "<{tag}{tag_style}>".format(tag=tag_name, tag_style=tag_style)
+        self.tag_close = "</{tag}>".format(tag=tag_name)
+
+
+    def append(self, content):
+        self.content.append(content)
+
+
+    def render(self, file_out, cur_ind = "", one_line_tag=False):
+        file_out.write(cur_ind + self.tag_open)
+        if not one_line_tag:
+            file_out.write('\n')
+
+        for content in self.content:
+            if type(content) == str:
+                file_out.write(content)
+            else:
+                content.render(file_out, cur_ind + "   ")
+            if not one_line_tag:
+                file_out.write('\n')
+
+        if not one_line_tag:
+            file_out.write('\n' + cur_ind)
+        file_out.write(self.tag_close)
+
+
+class Html(Element):
+    def __init__(self, content=None, **kwargs):
+        Element.__init__(self, tag_name="html", content=content, **kwargs)
+    
+    
+    def render(self, file_out, cur_ind = "", one_line_tag=False):
+        file_out.write('<!DOCtype html>\n')
+        Element.render(self, file_out, cur_ind=cur_ind, one_line_tag=one_line_tag)
+
+class Body(Element):
+    def __init__(self, content=None, **kwargs):
+        Element.__init__(self, tag_name="body", content=content, **kwargs)
+
+
+class Head(Element):
+    def __init__(self, content=None, **kwargs):
+        Element.__init__(self, tag_name="head", content=content, **kwargs)
+
+
+class OneLineTag(Element):
+    def __init__(self, tag_name, content=None, **kwargs):
+        Element.__init__(self, tag_name=tag_name, content=content, **kwargs)
+
+
+    def render(self, file_out, cur_ind = ""):
+        Element.render(self, file_out=file_out, cur_ind=cur_ind, one_line_tag=True)
+
+
+class P(OneLineTag):
+    def __init__(self, content=None, **kwargs):
+        OneLineTag.__init__(self, tag_name="p", content=content, **kwargs)
+
+
+class Title(OneLineTag):
+    def __init__(self, content=None, **kwargs):
+        OneLineTag.__init__(self, tag_name="title", content=content, **kwargs)
+
+
+class SelfClosingTag(Element):
+    def __init__(self, tag_name, content=None, **kwargs):
+        if content is not None:
+            raise TypeError
+        Element.__init__(self, tag_name=tag_name, content=None, **kwargs)
+        self.tag_open = self.tag_open[:-1] + "/>"
+
+
+    def append(self, content):
+        raise TypeError
+
+
+    def render(self, file_out, cur_ind = ""):
+        file_out.write(cur_ind + self.tag_open)
+
```

Comment:
Although Python will let you get away with it, you want to be very careful about changing the call signature when overriding methods in a subclass. This could be avoided here by moving the `OneLineTag` render logic into that subclass's `render` method as mentioned above and doing away with the `one_line_tag` parameter